homeassistant/components/osoenergy/__init__.py

```diff
@@ -78,4 +82,9 @@ def __init__(self, osoenergy: OSOEnergy, osoenergy_device: _T) -> None:
         """Initialize the instance."""
         self.osoenergy = osoenergy
         self.device = osoenergy_device
-        self._attr_unique_id = osoenergy_device.device_id
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, self.device.device_id)},
+            manufacturer=MANUFACTURER,
+            model=self.device.device_type,
+            name=self.device.device_name,
```

Comment:
```suggestion
            identifiers={(DOMAIN, osoenergy_device.device_id)},
            manufacturer=MANUFACTURER,
            model=osoenergy_device.device_type,
            name=osoenergy_device.device_name,
```

(personal preference, rename osoenergy_device to device, looks better ;) )